File: client-go/ColorContainer.go
package main

import (
	"strconv"
	"strings"
)

type ColorContainer struct {
	colors []int
}

func (c *ColorContainer) AddColor(color int) *ColorContainer {
	c.colors = append(c.colors, color)
	return c
}

func (c ColorContainer) Json() string {
	var stringParts []string
	stringParts = append(stringParts, `{"colors":[`)
	for _, c := range c.colors {
		stringParts = append(stringParts, strconv.Itoa(c))
		stringParts = append(stringParts, ",")
	}
	stringParts = stringParts[:len(stringParts)-1]
	stringParts = append(stringParts, "]}")
	return strings.Join(stringParts, "")
}

File: client-ruby/lib/color_container.rb
class ColorContainer
  attr_accessor :colors

  def initialize
    @colors = []
  end

  # @return [String]
  def json
    @colors.each { |c| raise TypeError unless c.is_a? Integer }
    str = '{"colors":['
    @colors.each do |c|
      str.concat(c.to_s, ',')
    end
    str.delete_suffix! ','
    str + ']}'
  end

  def add_color(color)
    raise TypeError unless color.is_a? Integer

    @colors.push(color)
  end

end
